src/babel/transformation/file/index.js

```diff
@@ -413,6 +413,30 @@ export default class File {
     return uid;
   }
 
+  addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
+    // Generate a unique name based on the string literals so we dedupe
+    // identical strings used in the program.
+    var stringIds = raw.elements.map(function(string) {
+      return string.value;
+    });
+    var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
```

Comment:
All good, no rush at all. Sorry for taking so long to get to this.

On Tuesday, 7 July 2015, Kevin Decker <notifications@github.com> wrote:

> In src/babel/transformation/file/index.js
> <https://github.com/babel/babel/pull/1878#discussion_r34029342>:
>
> > @@ -413,6 +413,30 @@ export default class File {
> >      return uid;
> >    }
> >
> > +  addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
> > +    // Generate a unique name based on the string literals so we dedupe
> > +    // identical strings used in the program.
> > +    var stringIds = raw.elements.map(function(string) {
> > +      return string.value;
> > +    });
> > +    var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
>
> Will look into it. (Currently on vacation so might take a little bit)
> … <#14e6838d0d0339d7_>
> On Tue, Jul 7, 2015 at 12:55 PM Sebastian McKenzie <
> notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> src/babel/transformation/file/index.js <
> https://github.com/babel/babel/pull/1878#discussion_r34028273>: > @@
> -413,6 +413,30 @@ export default class File { > return uid; > } > > +
> addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
> > + // Generate a unique name based on the string literals so we dedupe > +
> // identical strings used in the program. > + var stringIds =
> raw.elements.map(function(string) { > + return string.value; > + }); > +
> var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
> Yeah, a lookup table is probably the best bet. All that would be required
> actually would be a different argument given to generateUidIdentifier since
> this.declarations is already being used with the complete key. — Reply to
> this email directly or view it on GitHub <
> https://github.com/babel/babel/pull/1878/files#r34028273>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/pull/1878/files#r34029342>.
>


-- 
Sebastian McKenzie